app.ts

```diff
@@ -0,0 +1,230 @@
+				}
+				$("tbody").append(lastRow);
+			}
+			return records;
+		})
+		.catch(err => {
+			throw new Error('Error fetching records' + err);
+		});
+}
+
+/** recalculates the record range that includes inputValue */
+function searchRecordsAndResize(recordCount: number) {
+	let inputValue = $('#searchInput').val() as number;
+	if (inputValue < 0 || inputValue > recordCount) {
+		$('.modal').css('display', 'block');
+		$('.content').append(`<p>${inputValue} is not a number within the range. Please try a different number</p>`);
+		$('#page').empty().append(`Showing record: ${myClass.firstNumber} - ${myClass.lastNumber}`);
+		$('#searchInput').val('');
+		return;
+	}
+	let calculatedRows = calculatingRows();
+	/** divides the calculated max rows in half*/
+	const halfRange = Math.floor(calculatedRows / 2);
+	myClass.firstNumber = Math.max(0, inputValue - halfRange);
+	myClass.lastNumber = Math.min(recordCount, myClass.firstNumber + (calculatedRows - 1));
+	displayRecords(recordCount);
+}
+
+/** Navigates to the next set of records */
+function rightArrow(recordCount: number) {
+	$('#page').empty();
+	$('#searchInput').val('');
+	/** retrieves the last row */
+	const lastRow = document.querySelector("#recordsTable tbody .row:last-child");
+	/** checks if the last row exists */
+	if (lastRow) {
+		const cells = lastRow.querySelectorAll("td");
+		const lastRecord = [];
+		for (const cell of Array.from(cells)) {
+			lastRecord.push(cell.textContent || "");
+		}
+		/** determines te value in the last row */
+		const lastID = parseFloat(lastRecord[0]);
+		/** checks if the last value is within range */
+		if (0 <= lastID && lastID <= recordCount) {
+			$("tbody").empty();
+			/** calculates the first number of the page */
+			myClass.firstNumber = lastID + 1;
+			let calculatedRows = calculatingRows();
+			/** calculates the last number of the page */
+			myClass.lastNumber = myClass.firstNumber + (calculatedRows - 1);
+			displayRecords(recordCount);
+		}
+	}
+}
+
+function leftArrow(recordCount: number) {
+	$('#page').empty();
+	$('#searchInput').val('');
+	/** retrieves the first row */
+	const firstRow = document.querySelector("#recordsTable tbody .row:first-child");
+	if (firstRow) {
+		const cells = firstRow.querySelectorAll("td");
+		const firstRecord = [];
+		for (const cell of Array.from(cells)) {
+			firstRecord.push(cell.textContent || "");
+		}
+		/** determines te value in the first row */
+		const firstID = parseFloat(firstRecord[0]);
+		if (0 <= firstID && firstID <= recordCount) {
+			$("tbody").empty();
+			const calculatedRows = calculatingRows();
+			myClass.lastNumber = firstID - 1;
+			myClass.firstNumber = myClass.lastNumber - (calculatedRows - 1);
+			displayRecords(recordCount);
+		}
+	}
+}
+
+window.onload = () => {
+	createTable();
+	myClass.fetchRecordCount()
+		.then(count => {
+			let recordCount: number = count - 1;
+			displayRecords(recordCount);
+			handleResize(recordCount);
+			searchRecordsAndResize(recordCount);
```

Comment:
Would this not be to early to call this function? Not sure how the code inside would work with the input being empty. And this could also cause a double `displayRecords` which is bad. So at the very least, please ensure that a `displayRecords` does not happen.